### PR TITLE
Update conformance-tester for openshift 4.1.9

### LINKED ISSUE #4169: Set up simple e2e tests for openshift 4
**User Story**

As a developer of Kubermatic I want to make sure that my changes don't break Openshift

**Acceptance criteria**

* There are e2e tests for Openshift
* They verify that spinning up an openshift 4 controlplane works
* They verify that creating machines for an openshift 4 cluster works
* They verify that nodes created for openshift 4 join the cluster and become ready

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

This PR updates the conformance tester and some bits of the API to properly work with openshift 4.1.9.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4169 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: api/cmd/conformance-tests/main.go [hunk 1]
```diff
@@ -245,6 +246,13 @@ func main() {
 	opts.kubermaticClient = apiclient.New(httptransport.New(kubermaticAPIServerAddress, "", []string{"http"}), nil)
 	opts.kubermaticAuthenticator = httptransport.BearerToken(kubermaticServiceaAccountToken)
 
+	if opts.openshift {
+		opts.openshiftPullSecret = os.Getenv("OPENSHIFT_IMAGE_PULL_SECRET")
+		if opts.openshiftPullSecret == "" {
+			log.Fatal("testing openshift requires the `OPENSHIFT_IMAGE_PULL_SECRET` env var to be set")
```
Comment (xrstf) [inline api/cmd/conformance-tests/main.go]: Please uppercase `Testing`.

File: api/cmd/conformance-tests/runner.go [hunk 2]
```diff
@@ -357,8 +359,59 @@ func (r *testRunner) executeScenario(log *zap.SugaredLogger, scenario testScenar
...
+		func() error {
+			selector, err := labels.Parse(fmt.Sprintf("worker-name=%s", r.workerName))
+			if err != nil {
+				return fmt.Errorf("failed to parse selector: %v", err)
+			}
+			return wait.PollImmediate(5*time.Second, 15*time.Minute, func() (bool, error) {
+				clusterList := &kubermaticv1.ClusterList{}
+				listOpts := &ctrlruntimeclient.ListOptions{LabelSelector: selector}
+				if err := r.seedClusterClient.List(r.ctx, listOpts, clusterList); err != nil {
+					log.Errorw("listing clusters failed", zap.Error(err))
```
Comment (xrstf) [inline api/cmd/conformance-tests/runner.go]: Likewise please uppercase here and in line 408 and 413.

**Comment by alvaroaleman** [comment]
@xrstf updated the logs to be uppsercase, PTAL

/test pull-kubermatic-e2e-aws-openshift-4.1.9

**Pull request merged**